core/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -25,17 +25,22 @@
 import java.util.List;
 import java.util.Map;
 
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.env.Environment;
 import org.elasticsearch.index.analysis.AnalysisRegistry;
 import org.elasticsearch.plugins.IngestPlugin;
 import org.elasticsearch.script.ScriptService;
 import org.elasticsearch.threadpool.ThreadPool;
 
+import static org.elasticsearch.common.settings.Setting.Property;
+
 /**
  * Holder class for several ingest related services.
  */
 public class IngestService {
+    public static final Setting<Boolean> NEW_DATE_FORMAT =
+        Setting.boolSetting("ingest.new_date_format", false, Property.NodeScope, Property.Dynamic, Property.Deprecated);
```

Comment:
I think we should not set the the dynamic flag here, because this setting isn't actually updated at runtime in `PipelineStore` (it is an immutable property).  

Or instead we should make it a setting that can be setting that can be changed dynamically at runtime. (by using `ClusterSettings#addSettingsUpdateConsumer(...)` )